packages/ipfs/src/core/components/start.js

```diff
@@ -343,3 +356,48 @@ function createApi ({
 
   return api
 }
+
+/**
+ * @typedef {Object} CreateAPIConfig
+ * @property {APIManager} apiManager
+ * @property {Bitswap} [bitswap]
+ * @property {Block} [block]
+ * @property {BlockService} blockService
+ * @property {Config} config
+ * @property {StartOptions} constructorOptions
+ * @property {DAG} dag
+ * @property {Files} [files]
```

Comment:
```suggestion
 * @property {Bitswap} bitswap
 * @property {Block} block
 * @property {BlockService} blockService
 * @property {Config} config
 * @property {StartOptions} constructorOptions
 * @property {DAG} dag
 * @property {Files} files
```

any reasons to set those properties to be optional here?

Reply:
@Xmader I believe they were optional (as in they were not passed somewhere and in that case defaults were used). I might be mistaken, however I would much rather land this is as and then try to improve than keep making small changes here, because validating those takes quite a bit time that I would rather spend on other issues.

If you do want to help I would suggest maybe creating a separate pull that targets type-gen branch which:

1. Enables type-checking in the CI so we can see if things are still checking.
2. Make these kind of improvements there.

That way we know that those changes could be applied without having to fetch them and validing those locally. I hope this is reasonable.